src/main/java/muller/Muller.java

```diff
@@ -0,0 +1,63 @@
+package muller;
+
+import muller.command.Command;
+import muller.command.MullerException;
+import muller.parser.Parser;
+import muller.storage.Storage;
+import muller.task.TaskList;
+import muller.ui.Ui;
+
+/**
+ * The main class for the Muller application. This class initializes
+ * the UI, storage, and task list, and controls the main application flow.
+ */
+public class Muller {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    /**
+     * Constructor for the Muller class.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Muller(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (MullerException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+    /**
+     * Runs the main loop of the application, processing user commands
+     * until the exit command is given.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Parser p = new Parser();
```

Comment:
Is there a reason why the Parser instantiation is within the while loop? Perhaps it could be instantiated outside the while loop to prevent unnecessary object creation.